frontend/src/employee-frontend/components/unit/TabCalendar.tsx

```diff
@@ -0,0 +1,100 @@
+// SPDX-FileCopyrightText: 2017-2021 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+import React, { useContext, useState } from 'react'
+import styled from 'styled-components'
+import { ContentArea } from 'lib-components/layout/Container'
+import { useTranslation } from '../../state/i18n'
+import { defaultMargins } from 'lib-components/white-space'
+import { H1, H2 } from 'lib-components/typography'
+import { FixedSpaceRow } from 'lib-components/layout/flex-helpers'
+import { ChoiceChip } from 'lib-components/atoms/Chip'
+import Absences from '../absences/Absences'
+import GroupSelector from './tab-calendar/GroupSelector'
+import { UUID } from 'lib-common/types'
+import { UnitContext } from '../../state/unit'
+import LocalDate from 'lib-common/local-date'
+import UnitAttendanceReservationsView from './unit-reservations/UnitAttendanceReservationsView'
+import { useQuery } from '../../utils/useQuery'
+
+type Mode = 'week' | 'month'
+
+const TopRow = styled.div`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  width: 100%;
+  margin-bottom: ${defaultMargins.m};
+`
+
+const GroupSelectorWrapper = styled.div`
+  width: 500px;
+  margin-bottom: ${defaultMargins.m};
+`
+
+interface Props {
+  unitId: UUID
+}
+
+export default React.memo(function TabCalendar({ unitId }: Props) {
+  const { i18n } = useTranslation()
+  const { unitInformation } = useContext(UnitContext)
+  const [mode, setMode] = useState<Mode>('month')
+  const defaultGroup = useQuery().get('group')
+  const [groupId, setGroupId] = useState<UUID | 'no-group' | null>(defaultGroup)
```

Comment:
You might want to check that `defaultGroup` is a valid id. Also, the query parameter is not synced with the UI, which might get a bit confusing.

Reply:
I guess it should "work" even if it's not valid - the group just won't be found from the results and won't be shown. Or how should it behave? Redirect to no query param?

I'll try to implement the syncing. Do you happen to remember good example code for that?

Reply:
There's a `useSyncQueryParams` hook for that.

Reply:
I think the `group` value should be reset to the default value if it's not a valid selection.

Reply:
By the way, the select component dissappears if you pick the "no selection" option.

Reply:
had some trouble with infinite loops despite using memo, so added some extra conditions to that hook https://github.com/espoon-voltti/evaka/pull/1352/commits/04d8bb3d8e70b85a25bc1beffed5a29731cc90df

Reply:
The problems were probably because of the thing I just commented on. The extra conditions don't have an effect if you use `useMemo`.

Still, I got an idea for an improvement to `useSyncQueryParams` so you wouldn't have to use `useMemo` with it:
```
/**
 * Keeps current URL's query params up to date with the queryParams object
 */
export function useSyncQueryParams(queryParams: Record<string, string>) {
  const history = useHistory()
  const query = new URLSearchParams(queryParams).toString()
  useEffect(() => {
    const { pathname } = history.location
    const url = query ? `${pathname}?${query}` : pathname
    history.replace(url)
  }, [history, query])
}
```